contrib/bst-graph

```diff
@@ -55,6 +56,10 @@ def parse_args():
     return parser.parse_args()
 
 
+def escape(s):
+    return urllib.parse.quote_plus(s.encode())
```

Comment:
Is this the right function to use for escaping? Might [graphviz.escape()](https://graphviz.readthedocs.io/en/stable/api.html#graphviz.escape) be a better fit (I haven't tested it)?

Reply:
I see. What confused me is that graphviz doesn't actually expect URI/percent-encoded strings. As far as I can tell, this is using URI percent-encoding simply as a convenient way to generate unique identifiers that don't contain colons or backslashes. As labels are used, the identifiers don't really matter. (A cryptographic hash could be used instead but that would make the graphviz file less readable).

Can you please add a small comment that URI percent-encoding is used to generate unique identifiers that don't contain colons or backslashes?

Also, is the `.encode()` part actually necessary? `urllib.parse.quote_plus()` accepts strings and the default encoding is `utf-8`, so it seems to me that `.encode()` is redundant, or am I missing something?

Reply:
> Can you please add a small comment that URI percent-encoding is used to generate unique identifiers that don't contain colons or backslashes?

Added a small comment, also renamed the `escape` function to `unique_node_name`. Hopefully that makes the intent easier to grok for readers.

> Also, is the .encode() part actually necessary? urllib.parse.quote_plus() accepts strings and the default encoding is utf-8, so it seems to me that .encode() is redundant, or am I missing something?

Fixed